File: api/submit.py
"""维度之门 - 提交表单 API"""

from http.server import BaseHTTPRequestHandler
import json
import sqlite3
import os
import re
from datetime import datetime

DB_PATH = '/tmp/submissions.db'


def init_db():
    conn = sqlite3.connect(DB_PATH)
    c = conn.cursor()
    c.execute('''
        CREATE TABLE IF NOT EXISTS submissions (
            id          INTEGER PRIMARY KEY AUTOINCREMENT,
            name        TEXT    NOT NULL,
            company     TEXT    NOT NULL,
            phone       TEXT    NOT NULL,
            interests   TEXT,
            message     TEXT,
            created_at  TEXT    NOT NULL
        )
    ''')
    conn.commit()
    conn.close()


class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        content_length = int(self.headers.get('Content-Length', 0))
        body = self.rfile.read(content_length)

        try:
            data = json.loads(body) if body else {}
        except json.JSONDecodeError:
            data = {}

        name = (data.get('name') or '').strip()
        company = (data.get('company') or '').strip()
        phone = (data.get('phone') or '').strip()
        interests = data.get('interests', [])
        message = (data.get('message') or '').strip()

        # 校验
        if not name:
            return self._json(400, {'ok': False, 'error': '姓名不能为空'})
        if not company:
            return self._json(400, {'ok': False, 'error': '公司名称不能为空'})
        if not phone:
            return self._json(400, {'ok': False, 'error': '联系电话不能为空'})

        phone_clean = re.sub(r'[\s\-]', '', phone)
        if not re.match(r'^(\+?86)?1[3-9]\d{9}$|^\+?[0-9]{8,15}$', phone_clean):
            return self._json(400, {'ok': False, 'error': '请输入有效的手机号码'})

        if len(message) > 200:
            return self._json(400, {'ok': False, 'error': '留言不能超过200字'})

        interests_str = ','.join(interests) if isinstance(interests, list) else str(interests)
        created_at = datetime.now().strftime('%Y-%m-%d %H:%M:%S')

        init_db()
        conn = sqlite3.connect(DB_PATH)
        try:
            c = conn.cursor()
            c.execute(
                'INSERT INTO submissions (name, company, phone, interests, message, created_at) VALUES (?,?,?,?,?,?)',
                (name, company, phone, interests_str, message, created_at)
            )
            conn.commit()
            row_id = c.lastrowid
        finally:
            conn.close()

        return self._json(200, {'ok': True, 'id': row_id, 'message': '提交成功，我们将在24小时内联系您！'})

    def _json(self, status, data):
        self.send_response(status)
        self.send_header('Content-Type', 'application/json; charset=utf-8')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        self.wfile.write(json.dumps(data, ensure_ascii=False).encode('utf-8'))

    def do_OPTIONS(self):
        self._json(200, {})

File: functions/api/submit.js
/**
 * 维度之门 - 提交留资表单 API
 * Cloudflare Pages Function + D1
 * POST /api/submit
 */

export async function onRequestPost(context) {
  const { request, env } = context;
  const db = env.DB;

  // CORS headers
  const headers = {
    'Content-Type': 'application/json; charset=utf-8',
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Methods': 'POST, OPTIONS',
    'Access-Control-Allow-Headers': 'Content-Type',
  };

  try {
    const data = await request.json();

    const name = (data.name || '').trim();
    const company = (data.company || '').trim();
    const phone = (data.phone || '').trim();
    const interests = Array.isArray(data.interests) ? data.interests.join(',') : String(data.interests || '');
    const message = (data.message || '').trim();

    // Validation
    if (!name) return Response.json({ ok: false, error: '姓名不能为空' }, { status: 400, headers });
    if (!company) return Response.json({ ok: false, error: '公司名称不能为空' }, { status: 400, headers });
    if (!phone) return Response.json({ ok: false, error: '联系电话不能为空' }, { status: 400, headers });

    const phoneClean = phone.replace(/[\s\-]/g, '');
    const phoneRe = /^(\+?86)?1[3-9]\d{9}$|^\+?[0-9]{8,15}$/;
    if (!phoneRe.test(phoneClean)) {
      return Response.json({ ok: false, error: '请输入有效的手机号码' }, { status: 400, headers });
    }
    if (message.length > 200) {
      return Response.json({ ok: false, error: '留言不能超过200字' }, { status: 400, headers });
    }

    const createdAt = new Date().toISOString().replace('T', ' ').slice(0, 19);

    const result = await db.prepare(
      'INSERT INTO submissions (name, company, phone, interests, message, created_at) VALUES (?, ?, ?, ?, ?, ?)'
    ).bind(name, company, phone, interests, message, createdAt).run();

    return Response.json(
      { ok: true, id: result.meta.last_row_id, message: '提交成功，我们将在24小时内联系您！' },
      { status: 200, headers }
    );
  } catch (err) {
    return Response.json(
      { ok: false, error: '服务器错误，请稍后重试' },
      { status: 500, headers }
    );
  }
}

// Handle CORS preflight
export async function onRequestOptions() {
  return new Response(null, {
    headers: {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Methods': 'POST, OPTIONS',
      'Access-Control-Allow-Headers': 'Content-Type',
    },
  });
}
